pkg/auth/authorizer/abac/abac.go

```diff
@@ -100,13 +101,20 @@ func NewFromFile(path string) (policyList, error) {
 func (p policy) matches(a authorizer.Attributes) bool {
 	if p.subjectMatches(a) {
 		if p.Readonly == false || (p.Readonly == a.IsReadOnly()) {
-			if p.Resource == "" || (p.Resource == a.GetResource()) {
+			switch {
+			case p.NonResourcePath != "":
```

Comment:
Actually, we circle through a set of policies and attempt to match those against the request attributes here, so i'd argue it's correct to verify whether the policy is a non-resource policy first (in policies non-resource attributes invalidate ns/resource attributes) and then check if the request is matching, no?